test/ex_machina/ecto_test.ex

```diff
@@ -0,0 +1,115 @@
+defmodule ExMachina.EctoTest do
+  use ExUnit.Case, async: true
+
+  defmodule TestRepo do
+    def insert!(record) do
+      send self, {:created, record}
+      record
+    end
+  end
+
+  defmodule MyApp.Book do
+    defstruct title: nil, publisher: nil, __meta__: %{__struct__: Ecto.Schema.Metadata}, publisher_id: 1
+
+    def __schema__(:associations) do
+      [:publisher]
+    end
+  end
+
+  defmodule MyApp.ExMachina do
+    use ExMachina.Ecto, repo: TestRepo
+
+    def factory(:book) do
+      %MyApp.Book{
+        title: "Foo"
+      }
+    end
+
+    def factory(:user) do
+      %{
+        id: 3,
+        name: "John Doe",
+        admin: false
+      }
+    end
+
+    def factory(:article, attrs) do
+      %{
+        id: 1,
+        title: "My Awesome Article",
+        author_id: assoc(attrs, :author, factory: :user).id
+      }
+    end
+
+    def factory(:comment, attrs) do
+      %{
+        body: "This is great!",
+        article_id: assoc(attrs, :article).id
```

Comment:
I think we can leave it as is for now. Now that we've extracted it to it's own module, we can make another commit to allow for `article: assoc(attrs, :article)`